### PR TITLE
[WIP] Add timeout as parameter for Sprout client, Add remote worker option for replicated env

### PR DESCRIPTION
**Pull request by OpCode1300**
Fixing because the Sprout client can timeout when provisioning non-configured appliances under high load.

### CODE REVIEW

**Comment by dajoRH** [comment]
Lint report for commit 59e365e34f6f988c9b59223d604c8e1e405a6082:

`cfme/scripting/setup_env.py`:
- :bangbang: Line 42:101: **E501** *line too long (105 > 100 characters)*
- :bangbang: Line 91:101: **E501** *line too long (105 > 100 characters)*
- :bangbang: Line 113:101: **E501** *line too long (106 > 100 characters)*

Please, rectify these issues :smirk: .


**No commit flaws detected**.

*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit 647143276b90323839b709d08e6f8bffb5af4878:

`cfme/scripting/setup_env.py`:
- :bangbang: Line 77:101: **E501** *line too long (109 > 100 characters)*
- :bangbang: Line 121:94: **E231** *missing whitespace after ','*
- :bangbang: Line 121:101: **E501** *line too long (116 > 100 characters)*
- :bangbang: Line 135:101: **E501** *line too long (106 > 100 characters)*
- :bangbang: Line 137:101: **E501** *line too long (115 > 100 characters)*
- :bangbang: Line 142:101: **E501** *line too long (106 > 100 characters)*
- :bangbang: Line 150:101: **E501** *line too long (117 > 100 characters)*
- :bangbang: Line 152:55: **E231** *missing whitespace after ','*

Hmmm
:frowning: You might like to check the lint of your code :point_up: 


**No commit flaws detected**.

*CFME QE Bot*

File: cfme/scripting/setup_env.py [hunk 1]
```diff
@@ -14,14 +14,17 @@ def tot_time(string):
...
     tot = int(mtch.group('days') or 0) * 24 * 60
     tot += int(mtch.group('hours') or 0) * 60
-    tot += int(mtch.group('minutes')or 0)
+    tot += int(mtch.group('minutes') or 0)
     return tot
 
 
 def provision_appliances(count, cfme_version, provider, lease_time):
     sprout_client = SproutClient.from_config()
     apps, request_id = sprout_client.provision_appliances(version=str(cfme_version),
```
Comment (psav) [inline cfme/scripting/setup_env.py]: **REQUIRED:** Though it's been used elsewhere I really dislike this formatting and would prefer
```python
    apps, request_id = sprout_client.provision_appliances(
        version=str(cfme_version), count=count, preconfigured=False,
        lease_time=lease_time, provider=provider
    )
```

File: cfme/scripting/setup_env.py [hunk 2]
```diff
@@ -41,38 +44,37 @@ def setup_distributed_env(cfme_version, provider, lease):
...
-    print("Non-VMDB appliance provisioned and configured {}".format(ip1))
+    vmdb_appliance = apps[0]
+    node_appliance = apps[1]
+    port = (vmdb_appliance.address, '') if cfme_version >= "5.8" else (vmdb_appliance.address,)
+    command_set0 = ('ap', '', opt, '1', '1', 'y', '1', 'n', '1', pwd, TimedCommand(pwd, 360), '')
+    vmdb_appliance.appliance_console.run_commands(command_set0)
+    vmdb_appliance.wait_for_evm_service()
+    vmdb_appliance.wait_for_web_ui()
+    print("VMDB appliance provisioned and configured {}".format(vmdb_appliance.address))
+    command_set1 = ('ap', '', opt, '2', vmdb_appliance.address, '', pwd, '', '3') + port + \
```
Comment (psav) [inline cfme/scripting/setup_env.py]: **REQUIRED:** Please construct this in a way to remove the line continuation

File: cfme/scripting/setup_env.py [hunk 3]
```diff
@@ -41,38 +44,37 @@ def setup_distributed_env(cfme_version, provider, lease):
...
 @click.option('--lease', default='3h', help='set pool lease time, example: 1d4h30m')
 def setup_ha_env(cfme_version, provider, lease):
     lease_time = tot_time(lease)
     """multi appliance setup consisting of dedicated primary and standy databases with a single
     UI appliance."""
     print("Provisioning and configuring HA environment")
-    apps = provision_appliances(count=3, cfme_version=cfme_version, provider=provider,
-        lease_time=lease_time)
+    apps = provision_appliances(
+        count=3, cfme_version=cfme_version, provider=provider, lease_time=lease_time)
```
Comment (psav) [inline cfme/scripting/setup_env.py]: Last ```)``` on a new line

File: cfme/scripting/setup_env.py [hunk 4]
```diff
@@ -107,32 +109,56 @@ def setup_ha_env(cfme_version, provider, lease):
...
-    apps[1].set_pglogical_replication(replication_type=':remote')
+    remote_region_appliance.set_pglogical_replication(replication_type=':remote')
     print("Setup - Replication on global appliance")
-    apps[0].set_pglogical_replication(replication_type=':global')
-    apps[0].add_pglogical_replication_subscription(apps[1].address)
+    global_region_appliance.set_pglogical_replication(replication_type=':global')
+    global_region_appliance.add_pglogical_replication_subscription(remote_region_appliance.address)
+
+    if remote_worker:
+        port = (remote_region_appliance.address, '') \
```
Comment (psav) [inline cfme/scripting/setup_env.py]: **REQUIRED:** Please reconstructo remove the line continuation

**Comment by dajoRH** [comment]
Lint report for commit 9da244706519d75e326551bd20a14c8487d299d7:

`cfme/scripting/setup_env.py`:
- :bangbang: Line 46:62: **E231** *missing whitespace after ','*

Please, rectify these issues :smirk: .


**No commit flaws detected**.

*CFME QE Bot*

**Comment by dajoRH** [comment]
Would you mind rebasing this Pull Request against latest master, please? :trollface:
*CFME QE Bot*

**Comment by OpCode1300** [comment]
This PR is outdated and no longer needed.

**Pull request closed without merging**